### PR TITLE
Fix bug where signature is not verified for Java Key Store and add signColumnMasterkeyMetadata API to JKS and AKV

### PR DESCRIPTION
**Pull request by lilgreenbird**
adds the following new API to JKS and AKV to sign the column master key metadata and return generated signature:

    public byte[] signColumnMasterKeyMetadata(String masterKeyPath,
            boolean allowEnclaveComputations) throws SQLServerException {

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionAzureKeyVaultProvider.java [hunk 1]
```diff
@@ -865,6 +865,7 @@ private static String getVaultUrl(String masterKeyPath) {
     public boolean verifyColumnMasterKeyMetadata(String masterKeyPath, boolean allowEnclaveComputations,
             byte[] signature) throws SQLServerException {
         if (!allowEnclaveComputations) {
+            System.out.println("AKV verifyColumnMasterKeyMetadata returning false");
```
Comment (David-Engel) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionAzureKeyVaultProvider.java:868]: Oops. I assume this was from debugging. Remove.
```suggestion
```

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionJavaKeyStoreProvider.java [hunk 2]
```diff
@@ -113,6 +113,84 @@ public byte[] decryptColumnEncryptionKey(String masterKeyPath, String encryption
         return plainCEK;
     }
 
+    @Override
+    public boolean verifyColumnMasterKeyMetadata(String masterKeyPath, boolean allowEnclaveComputations,
+            byte[] signature) throws SQLServerException {
+
+        if (!allowEnclaveComputations) {
+            System.out.println("JKS verifyColumnMasterKeyMetadata returning false");
```
Comment (David-Engel) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionJavaKeyStoreProvider.java:121]: Oops. I assume this was from debugging. Remove.
```suggestion
```

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java [hunk 3]
```diff
@@ -382,8 +382,9 @@ protected Object[][] getContents() {
         {"R_AKVSignatureLengthError", "The specified encrypted column encryption key''s signature length: {0} does not match the signature length: {1} when using column master key (Azure Key Vault key) in {2}. The encrypted column encryption key may be corrupt, or the specified Azure Key Vault key path may be incorrect."},
         {"R_HashNull", "Hash should not be null while decrypting encrypted column encryption key."},
         {"R_NoSHA256Algorithm", "SHA-256 Algorithm is not supported."},
-        {"R_VerifySignature", "Unable to verify signature of the column encryption key."},
-        {"R_CEKSignatureNotMatchCMK", "The specified encrypted column encryption key signature does not match the signature computed with the column master key (Asymmetric key in Azure Key Vault) in {0}. The encrypted column encryption key may be corrupt, or the specified path may be incorrect."},
+        {"R_VerifySignatureFailed", "Unable to verify signature of the column encryption key."},
```
Comment (David-Engel) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java:385]:
```suggestion
        {"R_VerifySignatureFailed", "Unable to verify the signature of the column encryption key."},
```

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerSecurityUtility.java [hunk 4]
```diff
@@ -315,7 +315,8 @@ static void verifyColumnMasterKeyMetadata(SQLServerConnection connection, SQLSer
         }
 
         if (!provider.verifyColumnMasterKeyMetadata(keyPath, isEnclaveEnabled, cmkSignature)) {
-            throw new SQLServerException(SQLServerException.getErrString("R_VerifySignature"), null);
+            System.out.println("about to R_VerifySignatureFailed ");
```
Comment (David-Engel) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerSecurityUtility.java:318]: Oops. I assume this was from debugging. Remove.
```suggestion
```

**Review by tkyc — APPROVED** [review]

**Review by Jeffery-Wasty — APPROVED** [review]

**Pull request merged**